crates/query/src/scan/array_predicate.rs

```diff
@@ -409,4 +410,52 @@ impl ArrayPredicate for InList {
         let mask = sqd_polars::arrow::polars_boolean_to_arrow_boolean(&polars_mask);
         Ok(mask)
     }
-}
\ No newline at end of file
+}
+
+
+fn bitwise_and<const N: usize>(value: [u8; N], other: [u8; N]) -> [u8; N] {
```

Comment:
That's much better! Now let's optimize it even more by using `arrow::compute`-like computations. You can take a look at https://docs.rs/arrow/latest/arrow/compute/fn.binary.html

If it's expensive to get an arrow array here, let's just do the same by:
- using slices instead of `[u8; N]` so that no memory is copied
- using iterator methods like `fold` which are inlined and optimized very well

Reply:
experiments with arrow binary gave worse performance. fold and other method from `Iterator` trait gave 0 speed boost and i decided to keep initial approach.
i experimented locally on arm platform so i can imagine that on amd there could be performance gains... but until we sure i would prefer to keep code as simple as possible